components/utils/cachedrequestloader/src/main/java/org/mozilla/cachedrequestloader/CachedRequestLoader.java

```diff
@@ -41,16 +52,18 @@ public ResponseData getStringLiveData() {
     }
 
     private void loadFromCache() {
-        try {
-            new FileUtils.ReadStringFromFileTask<>(new FileUtils.GetCache(new WeakReference<>(context)).get(), subscriptionKey, stringLiveData, CachedRequestLoader::convertToPair).execute();
-        } catch (ExecutionException | InterruptedException e) {
-            e.printStackTrace();
-            Log.e(TAG, "Failed to open Cache directory when reading cached banner config");
-        }
+        ThreadUtils.postToMainThreadDelayed(() -> {
+            try {
+                new FileUtils.ReadStringFromFileTask<>(new FileUtils.GetCache(new WeakReference<>(context)).get(), subscriptionKey, stringLiveData, CachedRequestLoader::convertToPair).execute();
+            } catch (ExecutionException | InterruptedException e) {
+                e.printStackTrace();
+                Log.e(TAG, "Failed to open Cache directory when reading cached banner config");
+            }
+        }, Inject.getTestDelay(delayCacheLoad));
     }
 
     private void loadFromRemote() {
-        new RemoteLoadUrlTask(stringLiveData, this).execute(subscriptionUrl, userAgent, Integer.toString(socketTag));
+        ThreadUtils.postToMainThreadDelayed(() -> new RemoteLoadUrlTask(stringLiveData, this).execute(subscriptionUrl, userAgent, Integer.toString(socketTag)), Inject.getTestDelay(delayNetworkLoad));
```

Comment:
This is the same. Maybe use Injection to decide if we want to delay here ?
e.g. in Inject.java, under main:

```
void loadFromRemoteImp(){
    new RemoteLoadUrlTask(stringLiveData, this).execute(subscriptionUrl, userAgent, Integer.toString(socketTag));
}
```

and under test will be 
```
void loadFromRemoteImp(){
    ThreadUtils.postToMainThreadDelayed(() -> new RemoteLoadUrlTask(stringLiveData, this).execute(subscriptionUrl, userAgent, Integer.toString(socketTag)), Inject.getTestDelay(delayNetworkLoad));
}
```

Reply:
Fixed.